app/models/endpoint.rb

```diff
@@ -7,4 +7,8 @@ class Endpoint < ApplicationRecord
   validates :role, :uniqueness => { :scope => :source_id }
 
   acts_as_taggable
+
+  def base_url_path
+    URI::Generic.new(scheme, nil, host, port, nil, path, nil, nil, nil).to_s
```

Comment:
`URI::Generic.build` would be more readable since you can pass a hash.  Might just be me but I never liked long multiple positional arguments with mixed in nulls.

E.g. `URI::Generic.build(:scheme => scheme, :host => host, :port => port, :path => path)`